### ISSUE TITLE
[Bug]: Setting content breaks spacing, breaks active focused colors, and skips to the end of the line

### ISSUE THREAD

**Issue by JahnoelRondon** [ueberdosis/tiptap#4167]
### Code example

https://codesandbox.io/s/tiptap-react-editor-profanity-filter-n4x6tr?file=/src/App.js

### PROBLEM

Currently I have a censored filter for text that is working. The issue currently is the only way to apply this filter while a user is using tiptap is to set the content as they are writing onupdate of the editor. I managed to set the content using this filter to blur out profanities however when the content was set hitting the spacebar no longer triggered. Thats where I used setContents options ```js {preserveWhitespace: "full"}``` or ```js {preserveWhitespace: true} ``` and I was now able to hit the spacebar and see it in the editor. However...after one problem was solved. 2 others came with it. As a result of setting content onUpdate my buttons for color which previously were active would lose their focuse after any key press on the editor however the color styling would remain. Also another major issue now with setting content manually is the editor forcing the curcor to skip to the end of the text when trying to click on previous text and edit.

### What did you expect to happen?

I expect for the my buttons for textstyle to stay active and for me to be able to edit previous text on the same line rather than the editor forcing itself to the last line of text on manual setcontent updates for a developers stand point.

### What browser are you using?

Edge

### Anything to add? (optional)

Currently buttons like bold for checking if they are active work however I think this is because its a core part of tiptap whereass textstyle is an extension so something is wrong with textstyle in general. I had made a previous report for buttons not being able to unsetcolor when their current color is active. As for the spacing issue it seems as thought the problem lies within the setContent function itself. However the content it updating itself definitely does not seem to have the same functionality given to us via that function and or we are missing something in the docs to resolve this problem. 

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Which packages did you experience the bug in?

    "@tiptap/extension-color": "^2.0.0-beta.209",     "@tiptap/extension-placeholder": "^2.0.0-beta.209",     "@tiptap/extension-text-style": "^2.0.0-beta.209",     "@tiptap/react": "^2.0.0-beta.209",     "@tiptap/starter-kit": "^2.0.0-beta.209",

### What Tiptap version are you using?

2.0.0

**Comment by JahnoelRondon** [comment]
Dug into the abys of google and found a solution for the end of line issue. 

```js
let {from, to} = editor.state.selection;
editor.commands.setTextSelection({from, to});
```

from https://github.com/ueberdosis/tiptap/issues/2216

currently the only issue remainging is the colors being unfocused

**Comment by svenadlung** [comment]
Duplicate of #1818

**Comment by svenadlung** [comment]
@JahnoelRondon, it seems the "only" issue is the one with the color, so I'll close here. Let me know if I'm wrong.

**Comment by JahnoelRondon** [comment]
@svenadlung Its both the color parsing incorrectly and end of line options not being clear in the docs but both have been solved with fixes or workarounds in my post above. Also I've found that using editor.update function inside of useEffect is better than the useEditor hook for reasons I cant remember but they solve alot of the state and setting content issues ive had. This is good to be closed.
